From 558e6baedda1dda6c8dc968a74ce354150e543a6 Mon Sep 17 00:00:00 2001
From: Ashwin Sekhar T K <asekhar@marvell.com>
Date: Thu, 15 Dec 2022 19:28:48 +0530
Subject: [PATCH] enable cross compilation

Signed-off-by: Ashwin Sekhar T K <asekhar@marvell.com>
---
 lib/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Makefile b/lib/Makefile
index 4168b45..fa9e9a8 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -781,7 +781,8 @@ install: $(LIB_DIR)/$(LIBNAME)
 	install -m 0444 $(MAN2) $(MAN_DIR)
 	install -d $(LIB_INSTALL_DIR)
 ifeq ($(SHARED),y)
-	install -s -m $(LIBPERM) $(LIB_DIR)/$(LIBNAME) $(LIB_INSTALL_DIR)
+	install -s --strip-program=aarch64-marvell-linux-gnu-strip -m $(LIBPERM) \
+		$(LIB_DIR)/$(LIBNAME) $(LIB_INSTALL_DIR)
 else
 	# must not strip symbol table for static libs
 	install -m $(LIBPERM) $(LIB_DIR)/$(LIBNAME) $(LIB_INSTALL_DIR)
-- 
2.25.1

